core/types/uuid.go

```diff
@@ -15,9 +16,9 @@ type UUID [16]byte
 
 // NewUUIDV5 generates a uuidv5 from a byte slice.
 // This is used to deterministically generate uuids.
-func NewUUIDV5(from []byte) UUID {
-	u := uuid.NewSHA1(namespace, from)
-	return UUID(u)
+func NewUUIDV5(from []byte) *UUID {
+	u := UUID(uuid.NewSHA1(namespace, from))
+	return &u
```

Comment:
Why the change to pointer?  a 16 byte array is preferable to copy by value in general, but when it comes to the Valuer, that should really be a value receiver.

Reply:
Ah ok. I'll keep passing uuids by pointer, but make the valuer concrete